src/Plugin_Command.php

```diff
@@ -610,6 +606,12 @@ protected function install_from_repo( $slug, $assoc_args ) {
 	 * @alias upgrade
 	 */
 	public function update( $args, $assoc_args ) {
+		$all = Utils\get_flag_value( $assoc_args, 'all', false );
+
+		if ( ! ( $args = $this->check_optional_args_and_all( $args, $all ) ) ) {
+			return;
+		}
+
 		if ( isset( $assoc_args['version'] ) ) {
 			foreach ( $this->fetcher->get_many( $args ) as $plugin ) {
```

Comment:
The Fetcher class already goes through all this trouble, even checking whether the provided slugs are actually valid.

Wouldn't it be preferable to run the fetcher earlier, and then check whether it came back with a non-empty result?

Reply:
It's simpler this way. It's just checking that args have or have not been given, and giving an appropriate message. It's not interested in checking args, which is left to the existing code.